From: jorisc90 <joris@claassen.co>
Date: Thu, 25 Oct 2018 23:59:13 +0000
Subject: alsa-mixer: Update to support Arctis Pro Wireless headset

The Arctis 7 configuration can be used as is - the ALSA PCM and mixer
interfaces are the same.
---
 src/modules/alsa/mixer/paths/steelseries-arctis-input.conf     |  3 ++-
 .../alsa/mixer/paths/steelseries-arctis-output-mono.conf       | 10 +++++-----
 .../alsa/mixer/paths/steelseries-arctis-output-stereo.conf     | 10 +++++-----
 src/modules/alsa/mixer/profile-sets/90-pulseaudio.rules        |  5 +++++
 .../alsa/mixer/profile-sets/steelseries-arctis-usb-audio.conf  | 10 +++++-----
 5 files changed, 22 insertions(+), 16 deletions(-)

diff --git a/src/modules/alsa/mixer/paths/steelseries-arctis-input.conf b/src/modules/alsa/mixer/paths/steelseries-arctis-input.conf
index f3115ba..3fa36e9 100644
--- a/src/modules/alsa/mixer/paths/steelseries-arctis-input.conf
+++ b/src/modules/alsa/mixer/paths/steelseries-arctis-input.conf
@@ -13,7 +13,8 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
 
-; Steelseries Arctis 7 USB headset microphone path.
+; Steelseries Arctis 7 USB headset microphone path. Works also with Arctis Pro
+; Wireless.
 
 [General]
 description-key = analog-input-microphone-headset
diff --git a/src/modules/alsa/mixer/paths/steelseries-arctis-output-mono.conf b/src/modules/alsa/mixer/paths/steelseries-arctis-output-mono.conf
index 6795061..d8b24a2 100644
--- a/src/modules/alsa/mixer/paths/steelseries-arctis-output-mono.conf
+++ b/src/modules/alsa/mixer/paths/steelseries-arctis-output-mono.conf
@@ -13,11 +13,11 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
 
-; Steelseries Arctis 7 USB headset mono output path. The headset has two
-; output devices. The first one is mono, meant for voice audio, and the
-; second one is stereo, meant for everything else. The purpose of this
-; unusual design is to provide separate volume controls for voice and
-; other audio, which can be useful in gaming.
+; Steelseries Arctis 7 USB headset mono output path. Works also with Arctis Pro
+; Wireless. The headset has two output devices. The first one is mono, meant
+; for voice audio, and the second one is stereo, meant for everything else. The
+; purpose of this unusual design is to provide separate volume controls for
+; voice and other audio, which can be useful in gaming.
 
 [General]
 description-key = analog-output-headphones-mono
diff --git a/src/modules/alsa/mixer/paths/steelseries-arctis-output-stereo.conf b/src/modules/alsa/mixer/paths/steelseries-arctis-output-stereo.conf
index 4e10c80..fcc58a0 100644
--- a/src/modules/alsa/mixer/paths/steelseries-arctis-output-stereo.conf
+++ b/src/modules/alsa/mixer/paths/steelseries-arctis-output-stereo.conf
@@ -13,11 +13,11 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
 
-; Steelseries Arctis 7 USB headset stereo output path. The headset has two
-; output devices. The first one is mono, meant for voice audio, and the
-; second one is stereo, meant for everything else. The purpose of this
-; unusual design is to provide separate volume controls for voice and
-; other audio, which can be useful in gaming.
+; Steelseries Arctis 7 USB headset stereo output path. Works also with Arctis
+; Pro Wireless. The headset has two output devices. The first one is mono,
+; meant for voice audio, and the second one is stereo, meant for everything
+; else. The purpose of this unusual design is to provide separate volume
+; controls for voice and other audio, which can be useful in gaming.
 ;
 ; This path doesn't provide hardware volume control, because the stereo
 ; output is controlled by the PCM element with index 1, and currently
diff --git a/src/modules/alsa/mixer/profile-sets/90-pulseaudio.rules b/src/modules/alsa/mixer/profile-sets/90-pulseaudio.rules
index 264bfdb..5be9218 100644
--- a/src/modules/alsa/mixer/profile-sets/90-pulseaudio.rules
+++ b/src/modules/alsa/mixer/profile-sets/90-pulseaudio.rules
@@ -107,4 +107,9 @@ ATTRS{idVendor}=="041e", ATTRS{idProduct}=="322c", ENV{PULSE_PROFILE_SET}="sb-om
 ATTRS{idVendor}=="0bda", ATTRS{idProduct}=="4014", ENV{PULSE_PROFILE_SET}="dell-dock-tb16-usb-audio.conf"
 ATTRS{idVendor}=="1038", ATTRS{idProduct}=="1260", ENV{PULSE_PROFILE_SET}="steelseries-arctis-usb-audio.conf"
 
+# ID 1038:1294 is for Arctis Pro Wireless (which works with the Arctis 7
+# configuration).
+ATTRS{idVendor}=="1038", ATTRS{idProduct}=="1294", ENV{PULSE_PROFILE_SET}="steelseries-arctis-usb-audio.conf"
+
+
 LABEL="pulseaudio_end"
diff --git a/src/modules/alsa/mixer/profile-sets/steelseries-arctis-usb-audio.conf b/src/modules/alsa/mixer/profile-sets/steelseries-arctis-usb-audio.conf
index d3563a1..0dc0a9e 100644
--- a/src/modules/alsa/mixer/profile-sets/steelseries-arctis-usb-audio.conf
+++ b/src/modules/alsa/mixer/profile-sets/steelseries-arctis-usb-audio.conf
@@ -13,11 +13,11 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
 
-; Steelseries Arctis 7 USB headset. The headset has a microphone and two output
-; devices. The first output device is mono, meant for voice audio, and the
-; second one is stereo, meant for everything else. The purpose of this unusual
-; design is to provide separate volume controls for voice and other audio,
-; which can be useful in gaming.
+; Steelseries Arctis 7 USB and Arctis Pro Wireless USB headset. These headsets 
+; have a microphone and two output devices. The first output device is mono, 
+; meant for voice audio, and the second one is stereo, meant for everything 
+; else. The purpose of this unusual design is to provide separate volume 
+; controls for voice and other audio, which can be useful in gaming.
 ;
 ; See default.conf for an explanation on the directives used here.
 
